sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheckTest.java

```diff
@@ -0,0 +1,705 @@
+        final DefaultConfiguration checkConfig = createCheckConfig();
+        checkConfig.addAttribute("allowOverridingReturnValue", "true");
+        checkConfig.addAttribute("allowOverridingParameter", "true");
+
+        final String[] expected = {
+            "37:44: " + getCheckMessage(
+                    Jsr305AnnotationsCheck.MSG_PARAMETER_WITHOUT_NULLNESS_ANNOTATION, "e"),
+            "42:45: " + getCheckMessage(
+                    Jsr305AnnotationsCheck.MSG_PARAMETER_WITHOUT_NULLNESS_ANNOTATION, "e"),
+            "42:64: " + getCheckMessage(
+                    Jsr305AnnotationsCheck.MSG_PARAMETER_WITHOUT_NULLNESS_ANNOTATION, "e"),
+            "75:41: "
+                    + getCheckMessage(Jsr305AnnotationsCheck.MSG_PARAM_DEFINITIONS_WITH_CHECK, "e"),
+            "99:35: " + getCheckMessage(Jsr305AnnotationsCheck.MSG_PARAM_NONNULL_AND_NULLABLE, "e"),
+        };
+
+        verify(checkConfig, getPath("InputJsr305AnnotationsCheckWithParameter.java"), expected);
+    }
+
+    @Test
+    public void testAllowOverridingLambdas() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig();
+        checkConfig.addAttribute("allowOverridingReturnValue", "true");
+        checkConfig.addAttribute("allowOverridingParameter", "true");
+
+        final String[] expected = {
+            "45:5: " + getCheckMessage(
+                    Jsr305AnnotationsCheck.MSG_RETURN_WITHOUT_NULLNESS_ANNOTATION, "e"),
+        };
+
+        verify(checkConfig, getPath("InputJsr305AnnotationsCheckWithLambda.java"), expected);
+    }
+
+    @Test
+    public void testAllowOverridingInheritance() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig();
+        checkConfig.addAttribute("allowOverridingReturnValue", "true");
+        checkConfig.addAttribute("allowOverridingParameter", "true");
+
+        final String[] expected = {
+            "44:47: " + getCheckMessage(
+                    Jsr305AnnotationsCheck.MSG_PARAMETER_WITHOUT_NULLNESS_ANNOTATION, "e"),
+            "81:43: " + getCheckMessage(
+                    Jsr305AnnotationsCheck.MSG_PARAMETER_WITHOUT_NULLNESS_ANNOTATION, "e"),
+        };
+
+        verify(checkConfig, getPath("InputJsr305AnnotationsCheckWithInheritance.java"), expected);
+    }
+
+    @Test
+    public void testExcludePackage() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig();
+        checkConfig.addAttribute("excludePackages",
+                "com.github.sevntu.checkstyle.checks.coding.test");
+        checkConfig.addAttribute("allowOverridingParameter", "true");
+
+        final String[] expected = {};
+
+        verify(checkConfig, getPath("InputJsr305AnnotationsCheckWithParameter.java"), expected);
+    }
+
+    // ------------------ Unit tests (test particular functions) ------------------
+
+    @Test
+    public void testIsPackageExcluded() {
```

Comment:
The new way you are writting tests is the way we prefer. These type of tests we consider hacks and should be avoided as much as possible.
Can you convert these into normal tests? We don't mind having alot of input files.

Reply:
I think that those tests could be left out altogether. The test coverage without them was already high. In the code there are things like paranoia null checks which are never hit by the "normal tests" but of course identified as missing branches in cobertura. 

Where I found ways to use test input files for such corner cases I did it. I expect little more here. What would be your preference?

Reply:
Regression would show us if any NPEs that would happen if these extra protections were removed.
It is our preference to just remove the extra protections and fix them when user reports case on real code.

Reply:
OK. Done that. Another patch set is coming for it.